src/Microsoft.AspNetCore.TestHost/ResponseFeature.cs

```diff
@@ -13,10 +13,11 @@ internal class ResponseFeature : IHttpResponseFeature
     {
         private Func<Task> _responseStartingAsync = () => Task.FromResult(true);
         private Func<Task> _responseCompletedAsync = () => Task.FromResult(true);
+        private HeaderDictionary _headers = new HeaderDictionary();
 
         public ResponseFeature()
         {
-            Headers = new HeaderDictionary();
+            Headers = _headers;
             Body = new MemoryStream();
```

Comment:
Note to self, StatusCode and ReasonPhrase should also check HasStarted and throw